pkg/requirements/requirements.go

```diff
@@ -91,6 +91,46 @@ func CloneClusterRepo(g gitclient.Interface, gitURL string) (string, error) {
 	return dir, nil
 }
 
+func CloneClusterRepoSparse(g gitclient.Interface, gitURL string, cloneType string, sparseCheckoutPatterns []string) (string, error) {
```

Comment:
`sparseCheckoutPatterns` can be variadic.Same as how `gitclient.SparseCloneToDir()` accepts the arg

Reply:
Done